libs/fetcher.js

```diff
@@ -130,33 +135,59 @@ Request.prototype.clientConfig = function (config) {
 };
 
 /**
+ * capture meta data; capture stats for this request and pass stats data
+ * to options.statsCollector
+ * @method _captureMetaAndStats
+ * @param {Object} errData  The error response for failed request
+ * @param {Object} result  The response data for successful request
+ */
+Request.prototype._captureMetaAndStats = function (errData, result) {
+    var self = this;
+    var meta = (errData && errData.meta) || (result && result.meta);
+    if (meta) {
+        self.serviceMeta.push(meta);
+    }
+    var statsCollector = self._statsCollector;
+    if (typeof statsCollector === 'function') {
+        var err = errData && errData.err;
+        var stats = {
+            resource: self.resource,
+            operation: self.operation,
+            params: self._params,
+            statusCode: err ? err.statusCode : (result && result.meta && result.meta.statusCode || 200),
+            err: err,
+            time: Date.now() - self._startTime
+        };
+        statsCollector(stats);
+    }
+};
```

Comment:
We aligned `err` and `result` for the callback of `Request.prototype.end()`, not the `err` and `result` from `executeRequest` (private method for executing request).